src/solvers/dg/dg_2d.jl

```diff
@@ -266,7 +265,8 @@ function calc_volume_integral!(du::AbstractArray{<:Any,4}, u,
   end
 end
 
-@inline function split_form_kernel!(du, u, nonconservative_terms::Val{false}, equations,
+@inline function split_form_kernel!(du::AbstractArray{<:Any,4}, u,
```

Comment:
I recognized that we need this for dispatch to distinguish between different spatial dimensions. Currently, I use the `mesh` or the dimensions of the arrays `u, du`. We could also use the equations, but that feels a bit clumsy for me. Additionally, I thought we could dispatch on the array types, e.g. if someone implements GPU-specific stuff.

Reply:
OK. One question (though it's not really specific for this LOC): Why `<:Any` and not just `Any`? Since `Any` is abstract, is this semantically the same?

Reply:
Gotcha :+1: Thanks for the explanation including an illustrative code example!